### PR TITLE
fix: properly handle cpp templates

### PR DESCRIPTION
**Pull request by t0rr3sp3dr0**
This patch fixes an infinite recursion while processing C++ Template types.

Fixes: #463

### CODE REVIEW

**Comment by t0rr3sp3dr0** [comment]
This also fixes an issue where structs and unions were not being parsed properly, causing missing imports.

image

**Comment by t0rr3sp3dr0** [comment]
As a side effect of fixing struct and union parsing, types not handled properly by go-macho now show up as Obj-C Classes.

```objc
@class x1;

@interface NSDateComponents : NSObject <NSCopying, NSSecureCoding>

- (struct __CFDateComponents { struct __CFRuntimeBase { unsigned long long x0; _Atomic  x1; unsigned long long x2; } x0; struct __CFCalendar *x1; struct __CFTimeZone *x2; long long x3; long long x4; long long x5; long long x6; long long x7; long long x8; long long x9; long long x10; long long x11; long long x12; long long x13; long long x14; long long x15; long long x16; long long x17; long long x18; } *)_dateComponents;

**Comment by t0rr3sp3dr0** [comment]
The issue in go-macho should be address by this patch: https://github.com/blacktop/go-macho/pull/58

**Comment by blacktop** [comment]
@t0rr3sp3dr0 go-macho updated

**Comment by t0rr3sp3dr0** [comment]
We also need to get this one merged: https://github.com/blacktop/go-macho/pull/59

Noticed some weird `@class next;` stuff added to Foundation after this patch and that lead to another problem in go-macho.

**Comment by blacktop** [comment]
> We also need to get this one merged: [blacktop/go-macho#59](https://github.com/blacktop/go-macho/pull/59)
> 
> Noticed some weird `@class next;` stuff added to Foundation after this patch and that lead to another problem in go-macho.

done

File: internal/commands/macho/objc.go [hunk 1]
```diff
@@ -1179,5 +1194,55 @@ func (o *ObjC) nonBuiltInType(typ string) string {
...
+		if lDelimNext == -1 {
+			break
+		}
+
+		lDelimIt += lDelimNext + 1
+	}
+	rDelimIdx += 1
+
+	l := typ[:lDelimIdx]
+	m := typ[lDelimIdx+1 : rDelimIdx-1]
```
Comment (blacktop) [inline internal/commands/macho/objc.go:1221]: this line panics for certain frameworks

Comment (blacktop) [reply internal/commands/macho/objc.go:1221]: panic: runtime error: slice bounds out of range [11:2]

Comment (t0rr3sp3dr0) [reply internal/commands/macho/objc.go:1221]: Another go-macho bug: https://github.com/blacktop/go-macho/pull/60

**Comment by blacktop** [comment]
LGTM

File: internal/commands/macho/objc.go (hunk 1 above)
Comment (blacktop) [reply internal/commands/macho/objc.go:1221]: think we should armor this check and throw and error incase of some regression in the future?

Comment (t0rr3sp3dr0) [reply internal/commands/macho/objc.go:1279]: @blacktop, just added some checks to contextualSplit

**Pull request merged**